test/hotspot/jtreg/runtime/cds/appcds/SharedArchiveConsistency.java

```diff
@@ -397,36 +411,50 @@ public static void main(String... args) throws Exception {
         // modify jsa header, test should fail
         System.out.println("\n2. Corrupt header, should fail\n");
         modifyJsaHeader(copyFile(orgJsaFile, "corrupt-header"));
-        output = TestCommon.execCommon(execArgs);
+        output = shareAuto ? TestCommon.execAuto(execArgs) : TestCommon.execCommon(execArgs);
         output.shouldContain("The shared archive file has a bad magic number");
         output.shouldNotContain("Checksum verification failed");
+        if (shareAuto) {
+            output.shouldContain(HELLO_WORLD);
+        }
 
         // modify _jvm_ident, test should fail
         System.out.println("\n2a. Corrupt _jvm_ident, should fail\n");
         modifyJvmIdent(copyFile(orgJsaFile, "modify-jvm-ident"));
-        output = TestCommon.execCommon(execArgs);
+        output = shareAuto ? TestCommon.execAuto(execArgs) : TestCommon.execCommon(execArgs);
         output.shouldContain("The shared archive file was created by a different version or build of HotSpot");
         output.shouldNotContain("Checksum verification failed");
+        if (shareAuto) {
+            output.shouldContain(HELLO_WORLD);
+        }
 
-        // use the same archive as above, but run with -Xshare:auto
-        System.out.println("\n2b. Corrupt _jvm_ident run with -Xshare:auto\n");
-        output = TestCommon.execAuto(execArgs);
+        // use the same archive as above, but run with -Xshare:auto or -Xshare:on depends on shareAuto (reverse)
+        System.out.println("\n2b. Corrupt _jvm_ident run with " + (shareAuto ? "-Xshare:on\n" : "-Xshare:auto"));
+        output = shareAuto ? TestCommon.execCommon(execArgs) : TestCommon.execAuto(execArgs);
         output.shouldContain("The shared archive file was created by a different version or build of HotSpot");
-        output.shouldContain("Hello World");
+        if (!shareAuto) {
+          output.shouldContain("Hello World");
+        }
```

Comment:
Are lines 431-437 still necessary, now that both -Xshare:on and -Xshare:off are already executed for this test scenario.

Reply:
I was thinking of that either, the comment said using same archive. The other run with new created archive ---- though they have same contents. I could delete this part.